src/makeInfinityAdder.js

```diff
@@ -6,7 +6,23 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const adder = (number) => {
+    if (number !== undefined) {
+      sum += number;
+
+      return adder;
+    } else {
+      const result = sum;
```

Comment:
Is it necessary to use else statement if the body of the previous condition returns something?
```suggestion
      return adder;
    }
    
    const result = sum;
```